### ISSUE TITLE
vanilla/pure javascript documentation or reference and examples?

### ISSUE THREAD

**Issue by bannazz** [tensorflow/tfjs-core#750]
Hello, I'm considering taking the time and the effort learning the technology offered here but I'm not willing to learn typescript and can not use it. Actually I ditched angular as did a huge number of programmers because of the bad decision of changing its mother language to typescript and dropping the ".js" from the name. And that's Angular not a startup tech.

Deeplearn.js is supposedly a Javascript library yet I find 0 documentation/examples/demos or even notes of Javascript except on the home page and on the name of the library.
So where should I look?

If there is no docs/books...etc in pure js then, when are the docs going to be released?

Thanks in advance, please answer ASAP because it will make my decision and that of many other "typescript haters / javascript lovers" like me a lot easier knowing that info.

**Comment by erikwilson** [comment]
Typescript is compiled to Javascript, and you can use DeepLearn with any language that also compiles to Javascript, or just JS itself. 

This is probably a generic enough question that can be answered with something like: https://stackoverflow.com/questions/26427722/calling-properly-typescript-code-from-javascript

For my own obsolete example, https://github.com/erikwilson/deeplearn-graph-serializer uses ES7 and works with DL <0.5.

**Comment by nsthorat** [comment]
HI @bannazz 

We're now in tutorials mode, so expect to see tutorials in raw ES6 in the coming weeks. Currently, our code examples in the API reference are all es6: https://deeplearnjs.org/docs/api/

**Comment by dsmilkov** [comment]
Also we moved the tutorial [Training for ML Beginners](https://deeplearnjs.org/docs/tutorials/ml_beginners.html) to JavaScript (ES6).

**Comment by hpssjellis** [comment]
@bannazz 

I don't hate Typescript but I really like JavaScript. I also would like more examples in both Typescript AND JavaScript, since deeplearnjs is supposed to be a JavaScript library.

I have some very basic examples at https://hpssjellis.github.io/deeplearnjs-javascript-examples/ which has links to a github and a few videos. I would really like to partner with someone who has a better understanding of Machine Learning, as I want to teach it just need to understand it a bit better.

My twitter is @rocksetta

**Comment by nsthorat** [comment]
@tafsiri examples going forward will all be es6, TypeScript will only be used for internal development. We expect to write many more tutorials in the coming month or two :)

**Comment by bannazz** [comment]
Hello again,
I'm very glad to read such promising replies as well as I am sorry for my late reply;
I have been ill and off-line for about a week.

@erikwilson 

Your model saving (serializing) project is very useful; does deeplearn.js have a model saving feature built in yet?

if not, Any thoughts on how can I save the trained model in a neat way or have that feature implemented for the current version?

If you help me do it, we should share it with everyone because of how important and useful it will be to everyone who is going to use deeplearn.js.

@nsthorat 

I'm very happy that you responded this positively. Thank you. I was like, "alright I'm doing this" the moment I read your first reply.

Keep up your promises and your great intentions. :)

@dsmilkov

Great job daniel!
I'm looking forward to see more tutorials in raw js just like that one. :)

one in particular is, performance measures in 3 modes (not 2)
CPU, GPU without using textures, GPU with textures (if the library supports that)

does it support using textures like in gpu.js for example? (I will ask that question in a separate issue later)

@hpssjellis 

I'm a computer science researcher with two degrees in the field and I would love to work with you on those videos / tutorials / courses, here's my contact info:

twitter: @bannazz
facebook: https://facebook.com/bannazz
gmail: ahmed.hassanalbanna@gmail.com

Please do contact me. :)

Thank you all for the replies and your time keep up the good work.

**Comment by hpssjellis** [comment]
I will contact you @bannazz. We should probably make our own thread. Not sure what to call it. I want to know which ones of this list can be demonstrated in deeplearnjs now (and which in the future) and what is missing from the list (or should just be tossed from the list).

http://rocksetta.com/crowd-curriculum/default.php?playlist=9

Supervised Machine Learning, Regression Analysis, Classification, Unsupervised Machine Learning, Neural Networks, Algorithms, Linear Algorithms:, Gradient Descent, Linear Regression, Logistic Regression, Linear Discriminant Analysis, Non-Linear Algorithms: Classification And Regression Trees, Naive Bayes, K-Nearest Neighbors, Learning Vector Quantization, Support Vector Machines, Bagging and Random Forest, Boosting and AdaBoost



Video of how to use the Crowd Curriculum at
 https://www.youtube.com/watch?v=K1MmdGIMy-A&list=PL57Dnr1H_egukaDgFqwEnDVStd7Jktg1E&index=9

Trying to design a High School Level Javascript Machine Learning Curriculum. The hard part will be making as simple as possible example code for each of these curricular ideas.

**Comment by tafsiri** [comment]
I'm going to close this as we now have tensorflow/tfjs-examples with all javascript examples, we also have javascript code snippets in the docs and new tutorials that use JavaScript. Hope this helps!
